articles/clean-and-cleaner-code.md

```diff
@@ -0,0 +1,27 @@
+---
+authorName: Milan Milanovic
+authorGithubUsername: milanm
+issue: 59
+title: Clean and cleaner code
+---
+# {{page.title}}
+
+Being a developer in these days means to develop features and fix bugs for a customer sometimes in a fast pace, which may result in a badly maintainable code. On the other hand, we also want to make the most elegant solutions to some problems. This can produce code that machine can understand but people cannot so well. Such code can become a mayor issue for a company to maintain and use in a long run. 
+Our objective should be to write maintainable, understandable, simple and readable code, and we need to make an extra effort for this to happen. Code should be easy to change and understand to our colleagues, however this is not an easy task to achieve, especially on large projects. So we need to practice writing nice code and fail, fix it and repeat again until we achieve clean code. 
+This is the main way to create clean code, but we can achieve it  by applying some guidelines, principles and techniques, which will be shown below.
+*	Clearness - As a main guideline, the problem developer needs to solve is of the critical importance, however that must be understandable by a developer, often who didn’t write that code. 
+*	Focus - Good code should follow the Single responsibility principle (SRP), so that the code we write should only have one reason to change and to be small, or to say that is responsible of a single part of functionality. 
+*	Simplicity - We should always try to apply DRY (Don’t repeat yourself) rule. What this means is to reduce repetition of code, so that a single change does not produce many changes in code. In addition, we should follow the KISS principle (Keep It Simple Stupid), which force us not to complicate things unnecessary.
+*	Readability - In order to make code readable as much as we can, we can follow the YAGNI principlen(You Ain’t Gonna Need It), which implies that we shouldn’t write code upfront that is not currently needed.
+*	Naming - Naming things is one of the most important things in software development. Names should be expressive enough and clear that we immediately know what concept represents. This also represent a self-documenting approach.
```

Comment:
How about to emphasize this topic little more, due to importance? As you said, it is very important, so maybe adding a sentence or two would emphasize it a bit more?

Reply:
Yes, you're right. I adjusted this section.